oidc/viewsets.py

```diff
@@ -81,86 +85,103 @@ def logout(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:
             _("Unable to process OpenID connect logout request."),
         )
 
+    def _check_user_exists(self, user_data: dict) -> bool:
+        """
+        Helper function that checks if a user exists
+        """
+        if user_data.get(self.unique_user_filter_field):
+            field_value = user_data.get(self.unique_user_filter_field)
+            field = self.unique_user_filter_field + "__iexact"
+            return self.user_model.objects.filter(**{field: field_value}).count() > 0
+        return True
+
+    def generate_successful_response(self, request, user) -> HttpResponse:
+        """
+        Generates a success response for a successful Open ID Connect
+        Authentication request
+        """
+        response = HttpResponseRedirect(config.get("REDIRECT_AFTER_AUTH"))
+
+        if self.use_auth_backend:
+            login(
+                request, user, backend=self.auth_backend,
+            )
+
+        if self.use_sso:
+            sso_cookie = jwt.encode(
+                {"email": getattr(user, self.sso_cookie, "email")},
+                config.get("JWT_SECRET_KEY"),
+                config.get("JWT_ALGORITHM"),
+            )
+            response.set_cookie(
+                "SSO",
+                value=sso_cookie.decode("utf-8"),
+                max_age=self.cookie_max_age,
+                domain=self.cookie_domain,
+            )
+
+        return response
+
+    def map_claims_to_model_field(self, user_data) -> dict:
+        """
+        Maps claims to the appropriate field for model ingestion
+        """
+        data = {}
+        for k, v in user_data.items():
+            if k in self.user_creation_claims:
+                data[self.map_claim_to_model[k]] = v
+        return data
+
     @action(methods=["POST"], detail=False)
     def callback(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:
+        client = self._get_client(**kwargs)
         if self._get_client(**kwargs):
-            client = self._get_client(**kwargs)
-            if request.POST.get("id_token") or "username" in request.POST:
-                user_data = request.POST.copy().dict()
+            if request.POST.get("id_token"):
+                id_token = request.POST.get("id_token")
```

Comment:
Can we refactor this to the following:
```python
id_token = request.POST.get("id_token")
if id_token:
...
```
_Then we don't have to call `request.POST.get("id_token")` twice_

Reply:
Changed in the latest commits